### ISSUE TITLE
Training GREW when encountering a loss of nan

### ISSUE THREAD

**Issue by HUAFOR** [ShiqiYu/OpenGait#146]
I downloaded the latest OpenGait-Master and just modified the path to the dataset, evaluated gaitbase_da_grew on 8 nvidia 3090 GPUs, but found that I can't train it very soon, someone mentioned a similar problem before(https://github.com/ShiqiYu/OpenGait/issues/131), how should I adjust it? Thanks.

![image](https://github.com/ShiqiYu/OpenGait/assets/58834906/7b84b3f3-e49c-4e39-a1e8-5ac4f6bbc2b0)


[2023-07-31 11:50:40] [INFO]: -------- Train Pid List --------                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
[2023-07-31 11:50:40] [INFO]: [00001train, 00002train, ..., 20000train]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
[2023-07-31 11:52:12] [INFO]: {'lr': 0.1, 'momentum': 0.9, 'solver': 'SGD', 'weight_decay': 0.0005}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
[2023-07-31 11:52:12] [INFO]: {'gamma': 0.1, 'milestones': [80000, 120000, 150000], 'scheduler': 'MultiStepLR'}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
[2023-07-31 11:52:12] [INFO]: Parameters Count: 88.92230M                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
[2023-07-31 11:52:12] [INFO]: Model Initialization Finished!                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
[2023-07-31 11:54:01] [INFO]: Iteration 00100, Cost 223.11s, triplet_loss=0.8058, triplet_hard_loss=7.6296, triplet_loss_num=24446.3496, triplet_mean_dist=1.9661, softmax_loss=12.5998, softmax_accuracy=0.0007                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
[2023-07-31 11:55:43] [INFO]: Iteration 00200, Cost 102.25s, triplet_loss=0.5809, triplet_hard_loss=16.2685, triplet_loss_num=35945.6055, triplet_mean_dist=1.3897, softmax_loss=13.0619, softmax_accuracy=0.0001                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
[2023-07-31 11:57:26] [INFO]: Iteration 00300, Cost 103.20s, triplet_loss=0.3115, triplet_hard_loss=10.2011, triplet_loss_num=53099.5703, triplet_mean_dist=1.1834, softmax_loss=12.1675, softmax_accuracy=0.0000                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
[2023-07-31 11:59:07] [INFO]: Iteration 00400, Cost 100.48s, triplet_loss=0.5172, triplet_hard_loss=33.3988, triplet_loss_num=58240.5312, triplet_mean_dist=3.6161, softmax_loss=11.8381, softmax_accuracy=0.0004                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
[2023-07-31 12:00:49] [INFO]: Iteration 00500, Cost 102.30s, triplet_loss=0.3770, triplet_hard_loss=18.8925, triplet_loss_num=59438.4883, triplet_mean_dist=2.9147, softmax_loss=11.4598, softmax_accuracy=0.0002                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
[2023-07-31 12:02:34] [INFO]: Iteration 00600, Cost 104.94s, triplet_loss=1.1964, triplet_hard_loss=89.0976, triplet_loss_num=53815.1484, triplet_mean_dist=6.8021, softmax_loss=11.8048, softmax_accuracy=0.0001                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
[2023-07-31 12:06:44] [INFO]: Iteration 00700, Cost 250.17s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=41965.7617, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0000                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
[2023-07-31 12:08:24] [INFO]: Iteration 00800, Cost 99.88s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=40247.3242, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0002                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
[2023-07-31 12:10:04] [INFO]: Iteration 00900, Cost 99.55s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=40352.3672, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0001                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
[2023-07-31 12:11:44] [INFO]: Iteration 01000, Cost 100.48s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=40203.3945, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0002                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
[2023-07-31 12:13:25] [INFO]: Iteration 01100, Cost 100.37s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=40321.1289, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0000                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
[2023-07-31 12:15:04] [INFO]: Iteration 01200, Cost 99.83s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=40337.9453, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0000                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
[2023-07-31 12:16:44] [INFO]: Iteration 01300, Cost 99.95s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=40525.1484, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0001                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
[2023-07-31 12:18:25] [INFO]: Iteration 01400, Cost 100.18s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=40319.4531, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0002                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
[2023-07-31 12:20:05] [INFO]: Iteration 01500, Cost 100.08s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=40558.9766, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0000                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
[2023-07-31 12:21:45] [INFO]: Iteration 01600, Cost 100.43s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=40485.0352, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0002                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
[2023-07-31 12:23:25] [INFO]: Iteration 01700, Cost 100.04s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=40269.4961, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0001

**Comment by ChaoFan996** [comment]
Hello, so strange an error. 
Have you modified the officially provided cfg file except the GREW path ? 
Such as changing the batch size or learning rate.

**Comment by ChaoFan996** [comment]
that' ok, thanks for providing experiences.
